OctaneVSPlugin/Common/FieldsCache.cs

```diff
@@ -0,0 +1,424 @@
+
+        private static Metadata _defaultFieldsCache = null;
+        private static Metadata _persistedFieldsCache = null;
+
+        private FieldsCache()
+        {
+            DeserializeDefaultFieldsMetadata();
+            DeserializePersistedFieldsMetadata();
+        }
+
+        public static FieldsCache Instance
+        {
+            get
+            {
+                if (_fieldsCache == null)
+                {
+                    lock (_obj)
+                    {
+                        if (_fieldsCache == null)
+                        {
+                            _fieldsCache = new FieldsCache();
+                        }
+                    }
+                }
+
+                return _fieldsCache;
+            }
+        }
+
+        private static string Serialize()
+        {
+            using (MemoryStream memoryStream = new MemoryStream())
+            {
+                _serializer.WriteObject(memoryStream, _persistedFieldsCache);
+
+                memoryStream.Position = 0;
+                using (StreamReader sr = new StreamReader(memoryStream))
+                {
+                    return sr.ReadToEnd();
+                }
+            }
+        }
+
+        private static Metadata Deserialize(string json)
+        {
+            try
+            {
+                using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(json)))
+                {
+                    return (Metadata)_serializer.ReadObject(stream);
+                }
+            }
+            catch (Exception)
+            {
+                return new Metadata
+                {
+                    data = new Dictionary<string, HashSet<string>>(),
+                    version = 1
+                };
+            }
+        }
+
+        private static void DeserializeDefaultFieldsMetadata()
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+
+            using (Stream stream = assembly.GetManifestResourceStream("MicroFocus.Adm.Octane.VisualStudio.Resources.DefaultFields.json"))
+            using (StreamReader reader = new StreamReader(stream))
+            {
+                string result = reader.ReadToEnd();
+                _defaultFieldsCache = Deserialize(result);
+            }
+        }
+
+        private static void DeserializePersistedFieldsMetadata()
+        {
+            var persistedJson = OctanePluginSettings.Default.EntityFields;
+            _persistedFieldsCache = Deserialize(persistedJson) ?? new Metadata
+            {
+                data = new Dictionary<string, HashSet<string>>(),
+                version = 1
+            };
+        }
+
+        private static void PersistFieldsMetadata()
+        {
+            try
+            {
+                OctanePluginSettings.Default.EntityFields = Serialize();
+                OctanePluginSettings.Default.Save();
+            }
+            catch (Exception)
+            {
+            }
```

Comment:
Not sure if there is a logging mechanism, but if there is one maybe the exception should be logged.